libs/utils/utils.js

```diff
@@ -308,6 +351,40 @@ export const getFederatedUrl = (url = '') => {
   return url;
 };
 
+export function hasLanguageLinks(container, paths = LANGUAGE_BASED_PATHS) {
+  if (!container) return false;
+  const links = container.querySelectorAll('a');
```

Comment:
I remember @overmyheadandbody did some work to remove multiple iterations over links when transforming them. Would this have any effect on the work?

Reply:
It's certainly regressive, but I think for now the performance impact of iterating over links checking for language based sites is smaller than loading in the whole language map. At some point this might change though, and then I think we just load in the language map on every page load until every site is language based.